osidb/api_views.py

```diff
@@ -96,6 +97,123 @@ def perform_update(self, serializer) -> None:
         self.perform_save(serializer)
 
 
+class ValidateModelViewSetMixin(ModelViewSetMixin):
+    # mixin class defining mechanism of API validations
+    # which are unlike model validations tied to query itself
+    #
+    # TODO this is not a proper class doc string as it would generate
+    # a ton of unrelated API endpoint descriptions in OpenAPI schema
+
+    _mandatory_attrs = None
+    _mandatory_params = None
```

Comment:
query params should never be required, just add a common serializer field that is `write_only=True` and use that for storing the `bz_api_key` like done here: https://stackoverflow.com/a/52532194

Reply:
I will read through it. The query parameter will however somehow need to be required to successfully process the write request due to the Bugzilla audit record.

Reply:
@osoukup what I mean is that in DRF you can have fields that are not mapped in the model and only in the API, in this way you can have a serializer with a `bz_api_key` field that will be accepted in the JSON payload sent to the endpoint and then you can do whatever you want with it, without having it in the query param which I would argue is not even secure